api/querier/v1/querier.proto

```diff
@@ -150,3 +150,36 @@ message SelectSeriesRequest {
 message SelectSeriesResponse {
   repeated types.v1.Series series = 1;
 }
+
+message CallGraph {
+  repeated CallGraphNode nodes = 1;
+  repeated CallGraphEdge edges = 2;
+  // The total weight of the graph, used to compute percentages.
+  uint64 total_value = 3;
+  // The total number of nodes in the source full graph.
+  uint64 nodes_total = 4;
+  // The number of nodes not present in the graph.
+  uint64 nodes_dropped = 5;
+}
+
+message CallGraphNode {
+  // Values associated to this node. Self is exclusive to this node,
+  // total includes all descendents.
+  uint64 total = 1;
+  uint64 self = 2;
+  string name = 3;
+}
```

Comment:
Note that we use strings without indirection as in flame graph or pprof formats. The reason is that call graph nodes never duplicate; therefore, the names are meant to be unique.

In contrast to the call graph implementation in the pprof tool, we only keep the display name of the function. Function details such as file name, line number, versions, etc. should be retrieved via separate API (shared with flame graphs).